calculator/main.py

```diff
@@ -26,8 +26,18 @@ def perform_operation(calc, operation, params):
     :param params: Tuple containing the list of nums to operate with.
                    ie: (1, 2, 3, 4.5, -2)
     """
-    pass
-
+    # perform operation
+    for i in params:
+        if not isinstance(i, (int, float)):
+            raise InvalidParams("Given params are invalid.")
+    total = calc['operations'][operation](*params)
+    
+    # get the execution date-time and append in the history list
+    # double brackets since 'append' takes only one argument
+    calc['history'].append((str(datetime.now()), operation, params, total))
```

Comment:
Our team wasn't aware of the default formatting for datetime.now(), either. However, knowing about strftime is VERY helpful. :)